pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -354,6 +391,26 @@ func loadDataFromTemplate(clusterTemplateRevision *v3.ClusterTemplateRevision, c
 	return dataFromTemplate, nil
 }
 
+func convertRegistryMapToSliceInOrder(registryMap map[int]map[string]interface{}) []map[string]interface{} {
+	registries := make([]map[string]interface{}, len(registryMap))
+	for k, v := range registryMap {
+		registries[k] = v
+	}
+	return registries
+}
+
+func getIndexFromQuestion(question string) (int, error) {
+	var index int
+	var err error
+	re1, _ := regexp.Compile(`\[\d+\]$`)
+	target := re1.FindString(question)
```

Comment:
If there was no regex match, we would return `(0, nil)` - so caller will think index is 0. I think we should return an error in this case as no match by this regexp means we have malformed request.

Reply:
Good catch!  The `err` has a default value now, so `(0, nil)` will be returned only when 0 is the valid index.

Reply:
fixed.